opm/autodiff/FlowMainEbos.hpp

```diff
@@ -167,15 +168,12 @@ namespace Opm
                 int num_threads = std::min(4, num_cores);
                 omp_set_num_threads(num_threads);
             }
-#pragma omp parallel
-            if (omp_get_thread_num() == 0) {
```

Comment:
I think this part tries to make sure only one of the OpenMP threads write the message below, why did you remove that?

Reply:
the proposed code should be identical and it is quite a bit nicer, IMO. see @akva2's [comments](https://github.com/OPM/opm-simulators/pull/1316#discussion_r148505282)

Reply:
@atgeirr in the new code we avoid the fork into a parallel section, and thus there is only one thread around.

Reply:
Ok, I see now that it uses omp_get_max_threads() instead of omp_get_num_threads(), more questions:
 - Is it ok to call omp_get_max_threads() outside the parallel section? (Unlike omp_get_num_threads(), as said in the comment, that now should be removed).
 - Will it give equivalent results (i.e. can the user somehow choose a different number of threads than max)? (If so, why do both exist?)

Reply:
- omp_get_num_threads() returns *the number of currently active threads*. if you call it inside a  parallel section you get the number of spawned threads, if not you naturally get 1 always.
- opm_get_max_threads() returns the number of threads that will be used for the *next* parallel section.
- omp_set_num_threads() controls the returned value of omp_get_max_threads(), ie the number of threads that will be used in the next parallel section. you can never go above omp_get_thread_limit()

Reply:
Thanks for the thorough explanation, finally got it. I'll merge!